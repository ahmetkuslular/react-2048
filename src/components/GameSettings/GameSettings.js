import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

GameSettings.propTypes = {
  selectedTheme: PropTypes.string,
  changeTheme: PropTypes.func,
};

GameSettings.defaultProps = {
  selectedTheme: 'light',
};

function GameSettings({ selectedTheme, changeTheme }) {
  return (
    <Wrapper>
      <MoonButton onClick={changeTheme}>
        {selectedTheme === 'light' ? (
          <span role="img" aria-label={selectedTheme}>
            ðŸŒ’
          </span>
        ) : (
          <span role="img" aria-label={selectedTheme}>
            ðŸŒ”
          </span>
        )}
      </MoonButton>
      <Button>
        <Label>New Game</Label>
      </Button>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 10px;
`;

const MoonButton = styled.button`
  background-color: transparent;
  padding: 5px;
  text-transform: capitalize;
  font-size: 32px;
  outline: none;
  border: none;
  cursor: pointer;
`;

const Button = styled.button`
  min-width: 100px;
  text-align: center;
  justify-content: center;
  background-color: ${props => props.theme.primaryColor};
  padding: 5px;
  text-transform: capitalize;
  outline: none;
  border: none;
  cursor: pointer;
  border-radius: 5px;
`;

const Label = styled.div`
  color: ${props => props.theme.labelColor};
  font-weight: bold;
  font-size: 15px;
  padding: 5px;
`;

export default GameSettings;
