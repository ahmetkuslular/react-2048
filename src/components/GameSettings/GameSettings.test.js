import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import GameSettings from './GameSettings';

const changeTheme = jest.fn();
describe('components/GameSettings', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GameSettings theme="light" changeTheme={changeTheme}/>, div);
  });

  it('renders markup correctly for dark theme', () => {
    const tree = toJson(render(<GameSettings theme="dark" />));
    expect(tree).toMatchSnapshot();
  });

  it('renders markup correctly for light theme', () => {
    const tree = toJson(render(<GameSettings theme="light" />));
    expect(tree).toMatchSnapshot();
  });
});
