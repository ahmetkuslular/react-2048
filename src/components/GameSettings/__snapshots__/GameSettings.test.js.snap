// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GameSettings renders markup correctly for dark theme 1`] = `
<div
  class="sc-bdVaJa cdoCJP"
>
  <button
    class="sc-bwzfXH cludHl"
    id="moon"
  >
    <span
      aria-label="light"
      role="img"
    >
      ðŸŒ’
    </span>
  </button>
  <button
    class="sc-htpNat dMdHas"
    id="new-game"
  >
    <div
      class="sc-bxivhb lsVrT"
    >
      New Game
    </div>
  </button>
</div>
`;

exports[`components/GameSettings renders markup correctly for light theme 1`] = `
<div
  class="sc-bdVaJa cdoCJP"
>
  <button
    class="sc-bwzfXH cludHl"
    id="moon"
  >
    <span
      aria-label="light"
      role="img"
    >
      ðŸŒ’
    </span>
  </button>
  <button
    class="sc-htpNat dMdHas"
    id="new-game"
  >
    <div
      class="sc-bxivhb lsVrT"
    >
      New Game
    </div>
  </button>
</div>
`;
